### PR TITLE
Hotfix in I/O of Simulation and MCEventsProcessing

### PR DESCRIPTION
**Pull request by fhagemann**
When trying to run the [legend-julia-tutorial](https://github.com/legend-exp/legend-julia-tutorial), I encountered some bugs in the I/O.

- fbc9c23: The `json_string` is now stored in `Simulation` and not in `SolidStateDetector` (see #162)
- 024fe44: Replacing `:cartesian` and `:cylindrical` with actual types made the use of `Val(S)` obsolte.

### CODE REVIEW

File: src/Simulation/Simulation.jl [hunk 1]
```diff
@@ -53,7 +53,7 @@ function NamedTuple(sim::Simulation{T}) where {T <: SSDFloat}
     end
     wps_syms = Symbol.(wps_strings)
     nt = (
-        detector_json_string = NamedTuple(sim.detector.config_dict),
+        detector_json_string = NamedTuple(sim.config_dict),
```
[the remark this answers is not in the document]
Comment (fhagemann) [reply src/Simulation/Simulation.jl:56]: It is actually parsed as if it was a JSON dict (see [IO.jl L7](https://github.com/fhagemann/SolidStateDetectors.jl/blob/IO/src/IO/IO.jl#L7)), so I would keep `json` in the name

File: src/MCEventsProcessing/MCEventsProcessing_hdf5.jl [hunk 2]
```diff
@@ -8,7 +8,7 @@ function simulate_waveforms( mcevents::TypedTables.Table, s::Simulation{T},
     n_total_physics_events = length(mcevents)
     Δtime = T(to_internal_units(internal_time_unit, Δt)) 
     n_contacts = length(s.detector.contacts)
-    S = Val(get_coordinate_system(s.electric_potential.grid))
+    S = get_coordinate_system(s.electric_potential.grid)
```
[the remark this answers is not in the document]
Comment (fhagemann) [reply src/MCEventsProcessing/MCEventsProcessing_hdf5.jl:11]: It is needed in `_generate_waveform` in [MCEventsProcessing.jl L27](https://github.com/fhagemann/SolidStateDetectors.jl/blob/IO/src/MCEventsProcessing/MCEventsProcessing.jl#L27)

File: src/Simulation/Simulation.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (lmh91) [reply src/Simulation/Simulation.jl:56]: okay.

File: src/MCEventsProcessing/MCEventsProcessing_hdf5.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (fhagemann) [reply src/MCEventsProcessing/MCEventsProcessing_hdf5.jl:11]: I don't get your comment. Line 27 is part of `simulate_waveforms`....

[the remark this answers is not in the document]
Comment (lmh91) [reply src/MCEventsProcessing/MCEventsProcessing_hdf5.jl:11]: https://github.com/JuliaPhysics/SolidStateDetectors.jl/blob/024fe444d67561ec495c8d6434531c189f7e79d8/src/MCEventsProcessing/MCEventsProcessing_hdf5.jl

Line 27 is `HDF5.h5open(ofn, "w") do output`

[the remark this answers is not in the document]
Comment (fhagemann) [reply src/MCEventsProcessing/MCEventsProcessing_hdf5.jl:11]: MCEventsProcessing.jl, not MCEventsProcessing_hdf5.jl

[the remark this answers is not in the document]
Comment (lmh91) [reply src/MCEventsProcessing/MCEventsProcessing_hdf5.jl:11]: See above the file we are commenting on.

[the remark this answers is not in the document]
Comment (fhagemann) [reply src/MCEventsProcessing/MCEventsProcessing_hdf5.jl:11]: I see, we can remove it here if we want.

[the remark this answers is not in the document]
Comment (lmh91) [reply src/MCEventsProcessing/MCEventsProcessing_hdf5.jl:11]: Please, that would be nice.

**Review by lmh91 — APPROVED** [review]

**Pull request merged**